internal/provisioner/prometheus.go

```diff
@@ -71,15 +72,18 @@ func (p *prometheus) Create() error {
 		return err
 	}
 
-	logger := p.logger.WithField("prometheus-action", "create")
-
 	privateDomainName, err := p.awsClient.GetPrivateZoneDomainName(logger)
 	if err != nil {
 		return errors.Wrap(err, "unable to lookup private zone name")
 	}
 
 	app := "prometheus"
 	dns := fmt.Sprintf("%s.%s.%s", p.cluster.ID, app, privateDomainName)
+	_, err = net.LookupNS(dns)
```

Comment:
Hmm, I think it would be a lot better to check AWS for the correct record instead of doing a DNS lookup.

Reply:
That was my original intuition but this was just a much simpler approach. If you think it's worth the extra effort I can add a method to our AWS package to check through their APIs.

Reply:
Yeah, I figured that is what happened. I am fine with this as is for now, but I would like to have a ticket in the backlog to move to the AWS check instead if we don't want to handle it in this PR.